common/types.go

```diff
@@ -234,16 +237,302 @@ func init() {
 }
 
 // Address represents the 20 byte address of an Ethereum account.
-type Address [AddressLength]byte
+type InternalAddress [AddressLength]byte
+type ExternalAddress [AddressLength]byte
+
+type Address struct {
+	inner AddressData
+}
+
+type AddressData interface {
```

Comment:
Now that the `Address` type has exploded in complexity, a top level `AddressType`, an inner `AddressData` interface, and two address data types implementing the `AddressData` interface, _all_ with methods that should remain private, **I think we should move the address type into its own package**, similar to how `transaction.go` is its own module with a different `internal_tx.go`, `external_tx.go`, etc for all related data types.

Reply:
Fixed, please re-review. address.go, internal_address.go, external_address.go are all new